src/BotPlutusInterface/Contract.hs

```diff
@@ -229,13 +243,18 @@ handlePABReq contractEnv req = do
       WriteBalancedTxResp <$> writeBalancedTx @w contractEnv tx'
     AwaitSlotReq s -> AwaitSlotResp <$> awaitSlot @w contractEnv s
     AwaitTimeReq t -> AwaitTimeResp <$> awaitTime @w contractEnv t
-    CurrentPABSlotReq -> CurrentPABSlotResp <$> currentSlot @w contractEnv
+    CurrentNodeClientSlotReq -> CurrentNodeClientSlotResp <$> currentSlot @w contractEnv
     CurrentTimeReq -> CurrentTimeResp <$> currentTime @w contractEnv
     PosixTimeRangeToContainedSlotRangeReq posixTimeRange ->
       either (error . show) (PosixTimeRangeToContainedSlotRangeResp . Right)
         <$> posixTimeRangeToContainedSlotRange @w posixTimeRange
     AwaitTxStatusChangeReq txId -> AwaitTxStatusChangeResp txId <$> awaitTxStatusChange @w contractEnv txId
     AdjustUnbalancedTxReq unbalancedTx -> AdjustUnbalancedTxResp <$> adjustUnbalancedTx' @w @effs unbalancedTx
+    CurrentNodeClientTimeRangeReq -> do
+      -- TODO: This needs rework, we need to get the current slotLength from the eraSummaries
```

Comment:
Won't it break plutip that uses shorter slots?

Reply:
Ah, yeah it won't be completely correct.
I'll see if I can fix it